### ISSUE TITLE
Gauge visualization on small screens can cause frontend refresh on hover

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#15980]
**Describe the bug**
When viewing a Gauge visualization without labels on smaller resolution screens (below 1700px width), then hovering the ranges triggers the frontend to refresh, which on dashboards means it resets the filters and causes all cards to be queried again.
Regression since 0.39.0.

**To Reproduce**
1. Simple question > Orders > Sample Dataset > summarize Count and change visualization to Gauge - save as "Q1"
_It's possible to trigger in QB by making the viewport really tiny, but let's ignore that and continue to a more likely scenario_
2. Add "Q1" to a dashboard and save. Change the viewport to 1700px width, so the values are not shown anymore.
_This is just a test with a Gauge card with labels too, which doesn't trigger the problem, and a separate slow card with a filter to show it's causing a reset and new query_

https://user-images.githubusercontent.com/1447303/117576213-5087ab80-b0e5-11eb-9637-fb70b0f1a80d.mp4

<details><summary>Browser console is littered with errors, but this seems to be the cause</summary>

```
16:41:59.411 Error caught in <Visualization> TypeError: e is undefined
    getFriendlyName http://localhost:3001/app/dist/app-main.bundle.js?b395b892f139e711f915:5
    value http://localhost:3001/app/dist/app-main.bundle.js?b395b892f139e711f915:5
    value http://localhost:3001/app/dist/app-main.bundle.js?b395b892f139e711f915:5
    value http://localhost:3001/app/dist/app-main.bundle.js?b395b892f139e711f915:5
    Ya http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Ia http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    vs http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    vu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    gu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    uu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Vi http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    unstable_runWithPriority http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:495
    Wi http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Vi http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Ui http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    j http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    F http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Gt http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Xt http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    unstable_runWithPriority http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:495
    r http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Vt http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Ut http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    ft http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    on http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Za http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    yu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    vu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    gu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    uu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Vi http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    unstable_runWithPriority http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:495
    Wi http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Vi http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Ui http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    P http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    r http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Vt http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Ut http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    ft http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    on http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Za http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    yu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    vu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    gu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    uu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Vi http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    unstable_runWithPriority http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:495
    Wi http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Vi http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    Ui http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    iu http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    enqueueSetState http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:487
    setState http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:448
    _unlisten http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:495
    r http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:434
    l http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:434
    b http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:116
    b http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:116
    a http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:495
    a http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:495
    b http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:116
    b http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:116
    a http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:495
    l http://localhost:3001/app/dist/vendor.bundle.js?b395b892f139e711f915:434
 
Object { componentStack: "\n    in ChartTooltip\n    in div\n    in t\n    in Visualization\n    in Connect(Visualization)\n    in i\n    in Connect(i)\n    in div\n    in DashCard\n    in div\n    in t\n    in DraggableCore\n    in GridItem\n    in div\n    in GridLayout\n    in div\n    in t\n    in DashboardGrid\n    in div\n    in div\n    in div\n    in div\n    in div\n    in LoadingAndErrorWrapper\n    in t\n    in DashboardControls[t]\n    in Dashboard\n    in div\n    in t\n    in Unknown\n    in Title[]\n    in Title[Title[]]\n    in FitViewport\n    in DashboardApp\n    in Title[DashboardApp]\n    in Unknown\n    in UserIsAuthenticated(Component)\n    in Connect(UserIsAuthenticated(Component))\n    in MetabaseIsSetup(Connect(UserIsAuthenticated(Component)))\n    in Connect(MetabaseIsSetup(Connect(UserIsAuthenticated(Component))))\n    in div\n    in t\n    in ScrollToTop\n    in t\n    in App\n    in Title[App]\n    in RouterContext\n    in Router\n    in t\n    in DragDropContextProvider\n    in a" }
app-main.bundle.js:5:629636
```

</details>

**Information about your Metabase Installation:**
Tested 0.38.4 thru 0.39.1 and master. Does not cause this problem on 0.38.4, but the browser console is throwing a lot of errors on that version too.

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by nemanjaglumac** [comment]
Nitpick - this doesn't technically do a page reload but it does try to rerender. Noticed this while trying to assert that page refreshed while writing this repro:
Source: https://glebbahmutov.com/blog/detect-page-reload/

**Comment by marksull** [comment]
Stumbled across a workaround/root cause for this issue. I have a dashboard with 10 gauges, one was working fine, yet all others would trigger a refresh/reload.

If you add a label to each of the Gauge ranges, it will no longer cause the reload:

![image](https://user-images.githubusercontent.com/1705633/118436267-dfb74380-b723-11eb-83c0-7fa493bb0d14.png)

**Comment by alxnddr** [comment]
Closed by https://github.com/metabase/metabase/pull/16207

**Comment by drakedanner** [comment]
This error is still happening in 0.39.1

**Comment by flamber** [comment]
@drakedanner Yes, and it was fixed in 0.39.3. Latest release is now 0.40.0: https://github.com/metabase/metabase/releases
